docs/data/sdks/typescript-browser/migration.md

```diff
@@ -0,0 +1,497 @@
+    */
+    setup(config: BrowserConfig): Promise<undefined> {
+      this.config = config
+    }
+
+    /**
+    * execute() is called on each event instrumented
+    * example: client.track('New Event');
+    */
+    execute(event: Event): Promise<Event> {
+      event.library = 'my-library-name/1.0.0'
+      return Promise.resolve(event)
+    }
+  }
+  ```
+
+#### Defer initialization
+
+To defer initialization in `amplitude-js`, init must be called with `config.deferInitialization` set to true, and eventually calling `enableTracking()` to formalize initialization and send all enqueued events.
+
+=== "amplitude-js"
+  ```javascript
+  amplitude.getInstance().init(API_KEY, OPTIONAL_USER_ID, {
+    deferInitialization: true,
+  })
+
+  amplitude.getInstance().logEvent("Event 1")
+  amplitude.getInstance().logEvent("Event 2")
+  amplitude.getInstance().logEvent("Event 3")
+
+  amplitude.getInstance().enableTracking()
+  ```
+
+For `@amplitude/analytics-browser`, `init()` can be called at a later time than `track()`. All `track()` calls are then processed after initialization is completed.
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  amplitude.track("Event 1")
+  amplitude.track("Event 2")
+  amplitude.track("Event 3")
+
+  amplitude.init(API_KEY, OPTIONAL_USER_ID)
+  ```
+
+#### Web attribution
+
+In `amplitude-js`, web attribution is enabled by enabling the following configurations:
+
+* `config.includeGclid`
+* `config.includeFbclid`
+* `config.includeReferrer`
+* `config.includeUtm`
+
+Moving forward, the web attribution is controlled by a single configuration `config.attribution.disabled` which by default is set to `false` and captures all campaign parameters. This configuration collects the same campaign parameters supported in `amplitude-js`.
+
+#### Flush or `onExitPage`
+
+There are certain scenarios that necessiate sending events immediately, like navigating away from a page. This is a common scenario when tracking button clicks that directs the user to another page while sending event payload in batches.
+
+In `amplitude-js` this is done by using `onExitPage()` callback.
+
+=== "amplitude-js"
+  ```javascript
+  amplitude.getInstance().init(API_KEY, OPTIONAL_USER_ID, {
+    onExitPage: () => {
+      amplitude.sendEvents()
+    },
+  })
+  ```
+
+For `@amplitude/analytics-browser`, Amplitude recommends adding your own event listerer for `pagehide` event.
+
+=== "@amplitude/analytics-browser"
+  ```javascript
+  window.addEventListener('pagehide',
+    () => {
+      amplitude.setTransport('beacon') // Optional. Sets https transport to use `sendBeacon` API
+      amplitude.flush()
+    },
+  );
+  ```
+
+#### Callback
+
+For `amplitude-js`, two separate callback functions are passed for success and error. With `@amplitude/analytics-browser` supporting Promises (and async/await), the asynchronous methods like `track()`, `identify()`, `groupIdentify` return a custom promise interface.
```

Comment:
include `()` after `groupIdentify` for consistency.

Reply:
```suggestion
For `amplitude-js`, two separate callback functions are passed for success and error. With `@amplitude/analytics-browser` supporting Promises (and async/await), the asynchronous methods like `track()`, `identify()`, `groupIdentify()` return a custom promise interface.
```